app/src/test/java/com/pmatuki/wowpapers/usecases/GetDownloadApplyWallpaperTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.pmatuki.wowpapers.usecases
+
+import com.pmatuki.wowpapers.data.WallpapersRepositoryImpl
+import com.pmatuki.wowpapers.fake.WallpaperApplierFake
+import com.pmatuki.wowpapers.fake.WallpaperDownloaderFake
+import com.pmatuki.wowpapers.fake.WallpaperListSourceFake
+import com.pmatuki.wowpapers.usecases.apply.ApplyWallpaper
+import com.pmatuki.wowpapers.usecases.apply.WallpaperApplyResult
+import com.pmatuki.wowpapers.usecases.apply.WallpaperItemHolder
+import com.pmatuki.wowpapers.usecases.download.DownloadResult
+import com.pmatuki.wowpapers.usecases.download.DownloadWallpaper
+import com.pmatuki.wowpapers.usecases.get.GetWallpapers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class GetDownloadApplyWallpaperTest {
+
+    private val wallpaperListSourceFake = WallpaperListSourceFake(GetWallpapersTest.WALLPAPER_COUNT)
+
+    private val getWallpapers = GetWallpapers(WallpapersRepositoryImpl(wallpaperListSourceFake))
+
+    private val downloadWallpaper = DownloadWallpaper(WallpaperDownloaderFake())
+
+    private val applyWallpaper = ApplyWallpaper<ByteArray>(WallpaperApplierFake())
+
+    @Test
+    fun `test list, download second item and apply successfully`()  = runBlockingTest {
+        // List
+        val wallpaperList = getWallpapers()
+        Assert.assertTrue(wallpaperList.count() == GetWallpapersTest.WALLPAPER_COUNT)
+
+        // Download
+        val downloadResult = downloadWallpaper(wallpaperList[2].url)
+        Assert.assertTrue(downloadResult is DownloadResult.Success<*>)
+        val downloadSuccess = (downloadResult as DownloadResult.Success<*>)
+        Assert.assertNotNull(downloadSuccess.item)
+
+        // Apply
+        val applyResult = applyWallpaper(downloadSuccess.item as WallpaperItemHolder<ByteArray>)
+        Assert.assertTrue(applyResult is WallpaperApplyResult.Success)
+    }
+
+    @Test
+    fun `test list, error on download fifth item`()  = runBlockingTest {
+        // List
+        val wallpaperList = getWallpapers()
+        Assert.assertTrue(wallpaperList.count() == GetWallpapersTest.WALLPAPER_COUNT)
+
+        // Download with error
+        val downloadWallpaperWithError = DownloadWallpaper(WallpaperDownloaderFake().apply { doError() })
+        val downloadResult = downloadWallpaperWithError(wallpaperList[5].url)
+        Assert.assertTrue(downloadResult is DownloadResult.Error)
+        Assert.assertEquals((downloadResult as DownloadResult.Error).message, "Error downloading")
```

Comment:
This verification seems a little redundant. Is the error message important in your logic? If so, it is a good idea to keep it, otherwise I think this should be just a detail and doesn't need to be tested. 
Anyway, it is not wrong to keep it. 😊

Reply:
Indeed! The test will fail if developer changes this small detail, so I agree this should not be tested :).